src/backend/src/lib.rs

```diff
@@ -470,6 +475,38 @@ fn list_custom_tokens() -> Vec<CustomToken> {
     read_state(|s| s.custom_token.get(&stored_principal).unwrap_or_default().0)
 }
 
+#[update(guard = "caller_is_not_anonymous")]
```

Comment:
Is the plan to panic on error rather than return a Result?

Reply:
I have followed the same pattern as the other endpoints.

I would prefer to return an Error, shall I do that even when the pattern is not common in the other endpoints of the canister?

Reply:
In the frontend, you still need to use `try...catch` even if a `Result` is used. Therefore, unless there are specific details that must be included in the result, trapping errors, similarly to other end points of the backend, might fits better?

Reply:
I prefer to follow the current pattern unless otherwise specified.

There are different reasons why the validation might fail, but I don't think that any of them is something that the user could do something about. The user follows a flow and has no control over how the credential is built.